### PR TITLE
Increase 2 sec timeout up to 10 sec

### LINKED ISSUE #29: Operation timed out (2sec)
There are some other adhoc tasks failing when calling `tool_pdfpages\converter->convert_moodle_url_to_pdf()`:

```
Adhoc task failed: some\adhoc\task,URL to PDF conversion could not be completed.
Debug info:
Operation timed out (2sec)
```

`chrome-php` plugin maintainers advise simply increasing timeout:
* https://github.com/chrome-php/chrome/issues/353

### PR DESCRIPTION
**Pull request by golenkovm**
This is to fix #29

### CODE REVIEW

File: lang/en/tool_pdfpages.php [hunk 1]
```diff
@@ -36,6 +36,8 @@
 $string['settings:convertersheading'] = 'Converter settings';
 $string['settings:chromiumpath'] = 'Path to Chromium';
 $string['settings:chromiumpath_desc'] = 'The path to the Chrome/Chromium binary';
+$string['settings:responsetimeout'] = 'Response timeout';
```
Comment (dmitriim) [inline lang/en/tool_pdfpages.php]:
```suggestion
$string['settings:chromiumresponsetimeout'] = 'Response timeout';
```

File: lang/en/tool_pdfpages.php [hunk 2]
```diff
@@ -36,6 +36,8 @@
...
+$string['settings:responsetimeout_desc'] = 'Length of time in seconds to wait for response';
```
Comment (dmitriim) [inline lang/en/tool_pdfpages.php]:
```suggestion
$string['settings:chromiumresponsetimeout_desc'] = 'Length of time in seconds to wait for response from Chromium';
```

File: settings.php [hunk 3]
```diff
@@ -43,6 +43,11 @@
         get_string('settings:chromiumpath_desc', 'tool_pdfpages'),
         ''));
 
+    $settings->add(new admin_setting_configtext('tool_pdfpages/responsetimeout',
```
Comment (dmitriim) [inline settings.php]:
```suggestion
    $settings->add(new admin_setting_configtext('tool_pdfpages/chromiumresponsetimeout',
```

File: classes/converter_chromium.php [hunk 4]
```diff
@@ -103,7 +103,8 @@ protected function generate_pdf_content(moodle_url $proxyurl, string $filename =
             $page->navigate($proxyurl->out(false))->waitForNavigation();
             $pdf = $page->pdf($options);
 
-            return base64_decode($pdf->getBase64());
+            $timeout = 1000 * get_config('tool_pdfpages', 'responsetimeout');
```
Comment (dmitriim) [inline classes/converter_chromium.php]:
```suggestion
            $timeout = 1000 * helper::get_config($this->get_name() . 'responsetimeout');
```

File: settings.php [hunk 5]
```diff
@@ -43,6 +43,11 @@
...
+        get_string('settings:responsetimeout', 'tool_pdfpages'),
+        get_string('settings:responsetimeout_desc', 'tool_pdfpages'),
+        2, PARAM_INT));
```
Comment (dmitriim) [inline settings.php]: Why is it 2 by default?

File: classes/converter_chromium.php (hunk 4 above)
Comment (dmitriim) [reply classes/converter_chromium.php]: I'm confused by if we really need to multiply it by 1000?

File: settings.php (hunk 5 above)
Comment (golenkovm) [reply settings.php]: This is to make sure that the patch won't affect default behaviour.
2 sec is a default value in `chrome-php`: https://github.com/chrome-php/chrome/blob/f3deb4b31606309680bd587ea306499e462d0caa/src/Communication/ResponseReader.php#L112-L113

File: classes/converter_chromium.php (hunk 4 above)
Comment (golenkovm) [reply classes/converter_chromium.php]: `waitForResponse()` expects timeout to be in ms: https://github.com/chrome-php/chrome/blob/f3deb4b31606309680bd587ea306499e462d0caa/src/Communication/ResponseReader.php#L106-L115

File: classes/converter_chromium.php [hunk 6]
```diff
@@ -103,7 +103,8 @@ protected function generate_pdf_content(moodle_url $proxyurl, string $filename =
             $page->navigate($proxyurl->out(false))->waitForNavigation();
             $pdf = $page->pdf($options);
 
-            return base64_decode($pdf->getBase64());
+            $timeout = 1000 * helper::get_config($this->get_name() . 'responsetimeout');
```
Comment (golenkovm) [inline classes/converter_chromium.php:106]: Should this have `chromiumresponsetimeout` as well @dmitriim ?
```suggestion
            $timeout = 1000 * helper::get_config($this->get_name() . 'chromiumresponsetimeout');
```

File: settings.php (hunk 5 above)
Comment (dmitriim) [reply settings.php]: If it fails, should it be longer by default?

File: classes/converter_chromium.php (hunk 6 above)
Comment (dmitriim) [reply classes/converter_chromium.php:106]: Yeah. Look like should be responsetimeout and chromium will be prefixed in the helper.

File: settings.php (hunk 5 above)
Comment (golenkovm) [reply settings.php]: Does 10 sec sound good to you?

Comment (golenkovm) [reply settings.php]:
```suggestion
        10, PARAM_INT));
```

Comment (dmitriim) [reply settings.php]: if that fixes our issue, then sure.

File: settings.php [hunk 7]
```diff
@@ -43,6 +43,11 @@
         get_string('settings:chromiumpath_desc', 'tool_pdfpages'),
         ''));
 
+    $settings->add(new admin_setting_configtext('tool_pdfpages/chromiumresponsetimeout',
+        get_string('settings:responsetimeout', 'tool_pdfpages'),
```
Comment (dmitriim) [inline settings.php]: probably forgot to update those strings

**Comment by golenkovm** [comment]
Seems like everything is resolved @dmitriim

File: settings.php (hunk 7 above)
Comment (golenkovm) [inline settings.php]:
```suggestion
        get_string('settings:chromiumresponsetimeout', 'tool_pdfpages'),
```

File: settings.php [hunk 8]
```diff
@@ -43,6 +43,11 @@
...
+        get_string('settings:responsetimeout_desc', 'tool_pdfpages'),
```
Comment (golenkovm) [inline settings.php]:
```suggestion
        get_string('settings:chromiumresponsetimeout_desc', 'tool_pdfpages'),
```

**Pull request merged**